### ISSUE TITLE
[Bug]: ComboBox downshiftProps after selecting a item is not emitting onChange event

### ISSUE THREAD

**Issue by ashishkrz** [carbon-design-system/carbon#18212]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.72.0

### React version

18.3.1

### Description

There was an issue with Combobox with downshiftProps, which got fixed here https://github.com/carbon-design-system/carbon/issues/17027. But now the issue is , after the item gets selected from combobox, the onChange event is not happening.

### Reproduction/example

https://stackblitz.com/edit/github-v56j87-igdhfb?file=src%2FApp.jsx

### Steps to reproduce

Try selecting item from combobox, observe the onChange is not getting called after a selection:

onChange={(e) => { console.info('HELLO', e); }}

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

Power Virtual Server UI (PowerVS)

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey, do you see this issue on `1.72.0`?, the stackblitz you shared is on `1.71.1`

**Comment by ashishkrz** [comment]
@2nikhiltom I updated the package there, please check.

**Comment by ashishkrz** [comment]
@2nikhiltom Do we have any update on same?

**Comment by cknabe** [comment]
@Neues I think this is related to your change here: https://github.com/carbon-design-system/carbon/pull/18041

Passing in onStateChange in downshiftProps is overriding the onStateChange passed into useCombobox. The reason this breaks, is because onChange was moved from onInputValueChange and onSelectedItemChange into the new onStateChange. 

I reverted the changes in 18041 locally and everything works as expected. 

Without digging further into it, I guess there are a number of possibilities:
1. @ashishkrz or anyone else overriding onStateChange, can call onChange in their overriden code
2. A solution could be found where onChange is called outside of onStateChange
3. Revert 18041 and find a different solution for that issue

**Comment by Neues** [comment]
@cknabe Thanks for the tag and the detailed description of the issue. The solution that I came up with is to call downshiftProps.onStateChange within the existing onStateChange so that the onChange continues to work as expected. Let me know what you think.
https://github.com/carbon-design-system/carbon/pull/18441

**Comment by ashishkrz** [comment]
Hi @cknabe @Neues I didn't get what you are suggesting here? Do I have to change the way I am passing props **onStateChange** to downshiftProps.

**Comment by Neues** [comment]
> Hi [@cknabe](https://github.com/cknabe) [@Neues](https://github.com/Neues) I didn't get what you are suggesting here? Do I have to change the way I am passing props **onStateChange** to downshiftProps.

No, I was replying to @cknabe and describing the fix for the defect you reported in the PR I opened and linked to.
